include/envoy/thread_local/thread_local.h

```diff
@@ -228,5 +229,24 @@ class Instance : public SlotAllocator {
   virtual Event::Dispatcher& dispatcher() PURE;
 };
 
+struct MainThread {
```

Comment:
I'm wonder if https://github.com/envoyproxy/envoy/blob/master/include/envoy/thread/thread.h would be a better home? Or even `source/common/thread/thread.h`? If this is non-virtual and isn't needed inside `include/envoy` headers, then `source/common` is a better place.

Reply:
+1 on moving this class to `source/common/thread/thread.h`

Reply:
Thanks, I have moved the utility to thread_lib